src/operations/common_functions.ts

```diff
@@ -1,47 +1,5 @@
 import type { Document } from '../bson';
 import type { Collection } from '../collection';
-import type { Db } from '../db';
-import type { ReadPreference } from '../read_preference';
-import type { ClientSession } from '../sessions';
-
-/** @public */
-export interface IndexInformationOptions {
-  full?: boolean;
-  readPreference?: ReadPreference;
-  session?: ClientSession;
-}
-/**
- * Retrieves this collections index info.
- *
- * @param db - The Db instance on which to retrieve the index info.
- * @param name - The name of the collection.
- */
-export async function indexInformation(db: Db, name: string): Promise<any>;
-export async function indexInformation(
-  db: Db,
-  name: string,
-  options?: IndexInformationOptions
-): Promise<any>;
-export async function indexInformation(
-  db: Db,
-  name: string,
-  options?: IndexInformationOptions
-): Promise<any> {
-  if (options == null) {
-    options = {};
-  }
-  // If we specified full information
-  const full = options.full == null ? false : options.full;
-  // Get the list of indexes of the specified collection
-  const indexes = await db.collection(name).listIndexes(options).toArray();
-  if (full) return indexes;
-
-  const info: Record<string, Array<[string, unknown]>> = {};
-  for (const index of indexes) {
-    info[index.name] = Object.entries(index.key);
-  }
-  return info;
-}
 
 export function maybeAddIdToDocuments(
```

Comment:
I would be inspired to move this to utils and drop this file but I do not feel strongly at all.